### PR TITLE
fix: order the tests the same as tape

### PR DESCRIPTION
**Pull request by michaelfig**
Use the same test ordering as tape does.  Notably, we defer all the test suites, then execute them in order one by one in the main program.

This produces output much more like tape-promise.

### CODE REVIEW

**Comment by michaelfig** [comment]
I refactored the PR so that there aren't as many spurious changes (for simpler review).

@dckc, PTAL.

File: src/tape.js [hunk 1]
```diff
@@ -232,5 +232,44 @@ test.skip = function skip(label, htestOpt) {
...
 
-export default test;
-export { test };
+// If you want to defer the test suites and run them after
+// loading all the test modules, use:
+//   test.deferSuites();
+// and then later:
+//   await test.runDeferredSuites();
+
+let suitesToRun = undefined;
```
Comment (dckc) [inline src/tape.js]: sigh. more global mutable state (ambient authority).

Did you see warnings from xs tools (`xsl` I think) about this?

Is it not feasible to put this state inside `Harness`?

Comment (michaelfig) [reply src/tape.js]: Oh, sure!  I can put it in Harness.

**Comment by michaelfig** [comment]
For review, you should probably toggle off the whitespace diff option.

File: src/tape.js (hunk 1 above)
Comment (dckc) [reply src/tape.js]: thanks for putting it in Harness, but I still see it at module scope.
https://github.com/dckc/tape-xs/blob/b7b1a2920374946a676f96d68de109f3cdbbdea3/src/tape.js#L11

Is that still needed?

**Review by dckc — APPROVED** [review]
LGTM.

I'll leave merging to someone else until I get a chance to run it myself.

File: src/tape.js (hunk 1 above)
Comment (michaelfig) [reply src/tape.js]: Thanks, I removed the dead code.

**Pull request merged**